soccer/include/rj_vision_filter/VisionFilter.hpp

```diff
@@ -1,60 +1,86 @@
 #pragma once
 
-#include <ros2_temp/detection_frame_sub.h>
+#include <config_client/config_client.h>
+#include <rj_param_utils/ros2_param_provider.h>
+#include <rj_topic_utils/message_queue.h>
 
-#include <WorldState.hpp>
 #include <atomic>
 #include <mutex>
+#include <rclcpp/rclcpp.hpp>
+#include <rj_msgs/msg/detection_frame.hpp>
+#include <rj_msgs/msg/team_color.hpp>
+#include <rj_msgs/msg/world_state.hpp>
+#include <rj_utils/concurrent_queue.hpp>
 #include <rj_vision_filter/camera/CameraFrame.hpp>
 #include <rj_vision_filter/camera/World.hpp>
 #include <thread>
 #include <vector>
 
+namespace vision_filter {
+using TeamColorMsg = rj_msgs::msg::TeamColor;
+
 /**
  * Uses a seperate thread to filter the vision measurements into
  * a smoother velocity/position estimate for both the ball and robots.
  *
  * Add vision frames directly into the filter call the fill states functions
  * to push the newest estimates directly into the system state.
- * 
+ *
  * Note: There may be a 1 frame delay between the measurements being added
  * and the measurements being included in the filter estimate.
  */
-class VisionFilter {
+class VisionFilter : public rclcpp::Node {
 public:
+    using WorldStateMsg = rj_msgs::msg::WorldState;
+    using RobotStateMsg = rj_msgs::msg::RobotState;
+    using BallStateMsg = rj_msgs::msg::BallState;
+
     /**
      * Starts a worker thread to do the vision processing
      */
-    VisionFilter(Context* context);
-    ~VisionFilter();
+    VisionFilter(const rclcpp::NodeOptions& options);
 
-    void run();
+private:
+    /**
+     * @brief Runs prediction on all the Kalman Filters by calling out to
```

Comment:
For all the kalman filter references in this hpp/cpp, there's the kalman filter updates, then the merging of the best across each camera. I feel just mentioning the predict/kalman filters doesn't quite give the full story of what things happen and where the estimates come from.